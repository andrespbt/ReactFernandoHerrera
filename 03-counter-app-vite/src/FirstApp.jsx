import PropTypes from 'prop-types';
export const FirstApp = ({title, subTitle}) => {

  return (
    <>
        <h1>{title}</h1>
        <h2>My name is Andres</h2>
    </>
  )
}

FirstApp.propTypes = {
  title: PropTypes.string.isRequired

}