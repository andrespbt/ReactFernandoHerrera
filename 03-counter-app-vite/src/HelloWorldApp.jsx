export const HelloWorldApp = () => {
  return (
    <h1>Hola Mundo!!</h1>
  )
}

//Tarea
//Crear un componente llamado FirstApp