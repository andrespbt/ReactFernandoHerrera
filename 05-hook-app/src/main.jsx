import React from 'react';
import ReactDOM from 'react-dom/client';
import { TodoApp } from './08-useReducer/TodoApp';
// import { Padre } from './07-tarea-memo/Padre';
// import { CallbackHook } from './06-memos/CallbackHook';
// import { MemoHook } from './06-memos/MemoHook';
// import { Memorize } from './06-memos/Memorize';
// import { Layout } from './05-useLayoutEffect/Layout';
// import { FocusScreen } from './04-useRef/FocusScreen';
// import { FormWithCustomHook } from './02-useEffect/FormWithCustomHook';
// import { MultipleCustomHooks } from './03-examples/MultipleCustomHooks';
// import { SimpleForm } from './02-useEffect/SimpleForm';
// import { CounterWithCustomHook } from './01-useState/CounterWithCustomHook';
// import { CounterApp } from './01-useState/CounterApp';
// import { HooksApp } from './HooksApp';
// import './08-useReducer/into-reducer';
import './index.css';

ReactDOM.createRoot(document.getElementById('root')).render(
	// <React.StrictMode>
	<TodoApp />
	// </React.StrictMode>
);
