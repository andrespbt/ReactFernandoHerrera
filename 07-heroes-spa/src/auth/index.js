// eslint-disable-next-line import/export
export * from "./pages";
export * from "./context";
