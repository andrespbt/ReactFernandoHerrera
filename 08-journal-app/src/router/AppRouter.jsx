import { createBrowserRouter, Navigate, RouterProvider } from 'react-router-dom';
import { AuthRoutes, childAuthRoutes } from '../auth/routes/AuthRoutes';
import { useCheckAuth } from '../hooks/useCheckAuth';
import { childJournalRoutes, JournalRoutes } from '../journal/routes/JournalRoutes';
import { CheckingAuth } from '../ui';

export const AppRouter = () => {
  const status = useCheckAuth();

  const isAuthenticated = status === 'authenticated';

  if (status === 'checking') {
    return <CheckingAuth />;
  }

  const router = createBrowserRouter([
    {
      path: '/auth/*',
      element: isAuthenticated ? <Navigate to="/" /> : <AuthRoutes />,
      children: childAuthRoutes,
    },
    {
      path: '/',
      element: isAuthenticated ? <JournalRoutes /> : <Navigate to="/auth/login" />,
      children: childJournalRoutes,
    },
  ]);

  return (
    <>
      <RouterProvider router={router} />
    </>
  );
};
