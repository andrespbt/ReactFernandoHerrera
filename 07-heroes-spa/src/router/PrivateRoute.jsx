import { useContext } from "react";
import { Navigate, useLocation } from "react-router-dom";
import { AuthContext } from "../auth";
import PropTypes from "prop-types";

export const PrivateRoute = ({ children }) => {
  const { logged, user } = useContext(AuthContext);

  const { pathname, search } = useLocation();

  const lastPath = pathname + search;

  localStorage.setItem("lastPath", lastPath);
  localStorage.setItem("userName", user?.name);

  return logged ? children : <Navigate to="/login" />;
};

PrivateRoute.propTypes = {
  children: PropTypes.element,
};
