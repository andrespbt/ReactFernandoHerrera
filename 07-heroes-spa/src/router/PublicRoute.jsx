import { useContext } from "react";
import { Navigate } from "react-router-dom";
import { AuthContext } from "../auth";
import PropTypes from "prop-types";

export const PublicRoute = ({ children }) => {
  const { logged } = useContext(AuthContext);

  return !logged ? children : <Navigate to="/" />;
};

PublicRoute.propTypes = {
  children: PropTypes.element,
};
