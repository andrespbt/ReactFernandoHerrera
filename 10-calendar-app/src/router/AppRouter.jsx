import { createBrowserRouter, Navigate, RouterProvider } from 'react-router-dom';
import { LoginPage } from '../auth';
import { CalendarPage } from '../calendar';

export const AppRouter = () => {
  const authStatus = 'authenticated';

  const router = createBrowserRouter([
    {
      path: '/auth/*',
      element: authStatus === 'not-authenticated' ? <LoginPage /> : <CalendarPage />,
    },
    {
      path: '/*',
      element: <Navigate to="/auth/login" />,
    },
  ]);

  return (
    <>
      <RouterProvider router={router} />
    </>
  );
};
