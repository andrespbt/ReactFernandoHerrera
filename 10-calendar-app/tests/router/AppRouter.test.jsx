import { render, screen } from '@testing-library/react';
import { useAuthStore } from '../../src/hooks/useAuthStore';
import { AppRouter } from '../../src/router/AppRouter';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { store } from '../../src/store';
import { CalendarPage } from '../../src/calendar';

jest.mock('../../src/hooks/useAuthStore.js');

jest.mock('../../src/calendar', () => ({
  CalendarPage: () => <h1>CalendarPage</h1>,
}));

describe('tests on <AppRouter />', () => {
  beforeEach(() => jest.clearAllMocks());

  test('should show loading screen and call checkAuthToken', () => {
    const mockCheckAuthToken = jest.fn();

    useAuthStore.mockReturnValue({
      status: 'checking',
      checkAuthToken: mockCheckAuthToken,
    });

    render(<AppRouter />);

    expect(screen.getByLabelText('spinner')).toBeTruthy();

    expect(mockCheckAuthToken).toHaveBeenCalled();
  });

  test('should show login if user is not authenticated', () => {
    const mockCheckAuthToken = jest.fn();

    useAuthStore.mockReturnValue({
      status: 'not-authenticated',
      checkAuthToken: mockCheckAuthToken,
    });

    const { container } = render(
      <MemoryRouter initialEntries={['/auth2/asda/asda']}>
        <AppRouter />
      </MemoryRouter>
    );

    expect(screen.getAllByText('Login')).toBeTruthy();
    expect(container).toMatchSnapshot();
  });

  test('should show calendar if user is authenticated', () => {
    const mockCheckAuthToken = jest.fn();

    useAuthStore.mockReturnValue({
      status: 'authenticated',
      checkAuthToken: mockCheckAuthToken,
    });

    render(
      <MemoryRouter>
        <Provider store={store}>
          <AppRouter />
        </Provider>
      </MemoryRouter>
    );

    expect(screen.getAllByText('CalendarPage')).toBeTruthy();
  });
});
