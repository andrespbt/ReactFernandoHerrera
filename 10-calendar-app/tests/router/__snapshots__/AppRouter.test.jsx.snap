// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests on <AppRouter /> should show login if user is not authenticated 1`] = `
<div>
  <div
    class="container login-container"
  >
    <div
      class="row"
    >
      <div
        class="col-md-6 login-form-1"
      >
        <h3>
          Login
        </h3>
        <form>
          <div
            class="form-group mb-2"
          >
            <input
              autocomplete="username"
              class="form-control"
              name="loginEmail"
              placeholder="Email"
              type="text"
              value=""
            />
          </div>
          <div
            class="form-group mb-2"
          >
            <input
              autocomplete="current-password"
              class="form-control"
              name="loginPassword"
              placeholder="ContraseÃ±a"
              type="password"
              value=""
            />
          </div>
          <div
            class="form-group mb-2"
          >
            <input
              class="btnSubmit"
              type="submit"
              value="Login"
            />
          </div>
        </form>
      </div>
      <div
        class="col-md-6 login-form-2"
      >
        <h3>
          Register
        </h3>
        <form>
          <div
            class="form-group mb-2"
          >
            <input
              class="form-control"
              name="registerName"
              placeholder="Name"
              type="text"
              value=""
            />
          </div>
          <div
            class="form-group mb-2"
          >
            <input
              autocomplete="new-password"
              class="form-control"
              name="registerEmail"
              placeholder="Email"
              type="email"
              value=""
            />
          </div>
          <div
            class="form-group mb-2"
          >
            <input
              autocomplete="new-password"
              class="form-control"
              name="registerPassword"
              placeholder="Password"
              type="password"
              value=""
            />
          </div>
          <div
            class="form-group mb-2"
          >
            <input
              autocomplete="new-password"
              class="form-control"
              name="registerPassword2"
              placeholder="Repeat password"
              type="password"
              value=""
            />
          </div>
          <div
            class="form-group mb-2"
          >
            <input
              class="btnSubmit"
              type="submit"
              value="Create account"
            />
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
